src/common.c

```diff
@@ -142,47 +142,48 @@ static void print_source_error(FILE* out,
   }
 }
 
+static void print_error_header(FILE* out, WasmDefaultErrorHandlerInfo* info) {
+  if (info && info->header) {
+    switch (info->print_header) {
+      case WASM_PRINT_ERROR_HEADER_NEVER:
+        break;
+
+      case WASM_PRINT_ERROR_HEADER_ONCE:
+        info->print_header = WASM_PRINT_ERROR_HEADER_NEVER;
+        /* Fallthrough. */
+
+      case WASM_PRINT_ERROR_HEADER_ALWAYS:
+        fprintf(out, "%s:\n", info->header);
+        break;
+    }
+    /* If there's a header, indent the following message. */
+    fprintf(out, "  ");
+  }
+}
+
 void wasm_default_source_error_callback(const WasmLocation* loc,
                                         const char* error,
                                         const char* source_line,
                                         size_t source_line_length,
                                         size_t source_line_column_offset,
                                         void* user_data) {
-  print_source_error(stderr, loc, error, source_line, source_line_length,
-                     source_line_column_offset);
-}
-
-void wasm_default_assert_invalid_source_error_callback(
-    const WasmLocation* loc,
-    const char* error,
-    const char* source_line,
-    size_t source_line_length,
-    size_t source_line_column_offset,
-    void* user_data) {
-  fprintf(stdout, "assert_invalid error:\n  ");
-  print_source_error(stdout, loc, error, source_line, source_line_length,
-                     source_line_column_offset);
-}
-
-void wasm_default_assert_malformed_source_error_callback(
-    const WasmLocation* loc,
-    const char* error,
-    const char* source_line,
-    size_t source_line_length,
-    size_t source_line_column_offset,
-    void* user_data) {
-  fprintf(stdout, "assert_malformed error:\n  ");
-  print_source_error(stdout, loc, error, source_line, source_line_length,
+  WasmDefaultErrorHandlerInfo* info = user_data;
+  FILE* out = info && info->out_file ? info->out_file : stderr;
```

Comment:
Is it possible to enclose this in a macro/function, so that the (conditional) code doesn't have to be duplicated on every use.